### PR TITLE
Print generated GatewayClasses, TLSRoutes, TCPRoutes and ReferenceGrants.

### PR DESCRIPTION
**Pull request by dpasiukevich**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Print generated GatewayClasses, TLSRoutes, TCPRoutes and ReferenceGrants.

These resources are already present in the provider conversion logic but are not propagated to the printer yet. The PR fixes this.

**Which issue(s) this PR fixes**:

Related to the #100. Istio provider would generate TLSRoutes, TCPRoutes and ReferenceGrants for users.

**Does this PR introduce a user-facing change?**:

```release-note
Print generated GatewayClasses, TLSRoutes, TCPRoutes and ReferenceGrants in addition to Gateways and HTTPRoutes
```

### CODE REVIEW

File: cmd/print.go [hunk 1]
```diff
@@ -73,39 +72,103 @@ func (pr *PrintRunner) PrintGatewaysAndHTTPRoutes(cmd *cobra.Command, _ []string
...
+	for _, r := range gatewayResources {
+		resourceCount += len(r.GatewayClasses)
+		for _, gatewayClass := range r.GatewayClasses {
+			gatewayClass := gatewayClass
+			err := pr.resourcePrinter.PrintObj(&gatewayClass, os.Stdout)
+			if err != nil {
+				fmt.Printf("# Error printing %s GatewayClass: %v\n", gatewayClass.Name, err)
+			}
+		}
+	}
```
Comment (levikobi) [inline cmd/print.go:97]: It's just a nitpick, but is there any reason you repeated the outer for loop instead of having it once?

Comment (dpasiukevich) [reply cmd/print.go:97]: Kept the original logic: print all gateways from all providers, then all HTTPRoutes etc.

If it's in one loop, then the logic would change and we'd print all resources per provider.

Comment (levikobi) [reply cmd/print.go:97]: 100%

**Review by levikobi — APPROVED** [review]
Thanks @dpasiukevich!
/lgtm

**Comment by LiorLieberman** [comment]
Thanks @dpasiukevich !
/approve

**Pull request merged**